File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res{""};
        char curr_c{' '};
        int count{0};

        for (char c : s) {
            if (c != curr_c) {
                res += c;
                curr_c = c;
                count = 1;
            } else {
                if (count != 2) {
                    res += c;
                    count += 1;
                }
            }
        }
        return res;
    }
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res = ''
        curr_char = ''
        curr_count = 0
        for i in range(len(s)):
            if s[i] == curr_char:
                if curr_count >= 2:
                    continue
                else:
                    res += s[i]
                    curr_count += 1
            else:
                res += s[i]
                curr_char = s[i]
                curr_count = 1
                
        return res
                

